src/swupd/hash.go

```diff
@@ -0,0 +1,28 @@
+package swupd
+
+type hashval int
```

Comment:
I really don't agree with aliasing/typedef-ing primitive types unless we are doing obscure, object specific things in functions with them. I'm all for defining new types where necessary but this can go overboard easily imo and I want to point it out now... i.e the comment doesn't make sense  when it says it returns an index (simple number) but the func returns a type hashval instead, just return an int.